docs/dev.md

```diff
@@ -64,6 +70,71 @@ dep ensure
 /go/bin/deepcopy-gen --output-base=/go/src --input-dirs=github.com/kubeflow/experimental-kvc/pkg/apis/cr/v1/...
 go build -gcflags "-N -l" github.com/kubeflow/experimental-kvc
 ```
+### Developing on a workstation
+This is the preferred method for developers who don't want to run `docker` locally and or don't mind setting up
+`Go` development environment on their workstation.
+
+Prior to using this method please ensure you have `go 1.9.2` or better development environment setup and running.
+
+For developing on `Linux` add the following lines to your `.bashrc`
+```bash
+export GOROOT="/usr/lib/go-1.9"
+export GOPATH="$HOME/go"
+export PATH="$PATH:$GOROOT/bin:$GOPATH/bin"
+```
+
+and for `MacOSX` add these lines to your `.bashrc`:
+ ```bash
+export GOROOT="/usr/local/opt/go/libexec"
+export GOPATH=$HOME/go
+export PATH=$PATH:$GOROOT/bin:$GOPATH/bin
+```
+in both cases, install `mercurial` either using `yum` or `apt-get` or `brew`.
```

Comment:
Why would we need `mercurial`? I don't think we need it. We only need stuff that's in the dockerfile of the golang docker [here](https://github.com/kubeflow/experimental-kvc/blob/master/docker/golang/Dockerfile). Does it have mercurial too?

Reply:
We do, that's where `hg` is coming from, otherwise we get:
```
$ make dep-ensure 
dep ensure
grouped write of manifest, lock and vendor: error while writing out vendor tree: failed to write dep tree: failed to export bitbucket.org/ww/goautoneg: no valid source could be created: 
	failed to set up sources from the following URLs:
https://bitbucket.org/ww/goautoneg
: hg is not installed: 
	failed to set up sources from the following URLs:
ssh://hg@bitbucket.org/ww/goautoneg
: hg is not installed: 
	failed to set up sources from the following URLs:
http://bitbucket.org/ww/goautoneg
: hg is not installed: 
	failed to set up sources from the following URLs:
https://bitbucket.org/ww/goautoneg
: remote repository at https://bitbucket.org/ww/goautoneg does not exist, or is inaccessible: remote: Not Found
fatal: repository 'https://bitbucket.org/ww/goautoneg/' not found
: exit status 128
	failed to set up sources from the following URLs:
ssh://git@bitbucket.org/ww/goautoneg
: remote repository at ssh://git@bitbucket.org/ww/goautoneg does not exist, or is inaccessible: not a git repository.
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
: exit status 128
	failed to set up sources from the following URLs:
git://bitbucket.org/ww/goautoneg
: remote repository at git://bitbucket.org/ww/goautoneg does not exist, or is inaccessible: fatal: read error: Connection reset by peer
: exit status 128
	failed to set up sources from the following URLs:
http://bitbucket.org/ww/goautoneg
: remote repository at http://bitbucket.org/ww/goautoneg does not exist, or is inaccessible: remote: Not Found
fatal: repository 'http://bitbucket.org/ww/goautoneg/' not found
: exit status 128
make: *** [dep-ensure] Error 1
```

By the way in the docker we have:
```
root@297f16c40c8c:~# dpkg -l | grep merc
ii  mercurial                  4.0-1+deb9u1                   amd64        easy-to-use, scalable distributed version control system
ii  mercurial-common           4.0-1+deb9u1                   all          easy-to-use, scalable distributed version control system (common files)
```